federation/cmd/federated-apiserver/app/server.go

```diff
@@ -390,9 +240,8 @@ func Run(s *options.APIServer) error {
 
 	var serviceAccountGetter serviceaccount.ServiceAccountTokenGetter
 	if s.ServiceAccountLookup {
-		// If we need to look up service accounts and tokens,
-		// go directly to etcd to avoid recursive auth insanity
-		serviceAccountGetter = serviceaccountcontroller.NewGetterFromStorageInterface(etcdStorage)
+		// TODO: Maybe do not expose this option in genericapiserver, if most servers do not need it?
```

Comment:
Ah, so this is about authenticating requests? IMO:

* genericapiserver should, by default, defer to the main apiserver.
* genericapiserver shouldn't offer direct-to-etcd shortcuts at all, because I expect that to just plain not work in most setups.
* ubernetes apiserver probably *won't* accept service accounts from its federated clusters at all. This becomes a much larger discussion about auth for ubernetes which probably needs to involve @erictune and @quinton-hoole if not more people.

For the moment, we can *definitely* say that ubernetes won't have direct access to its federated cluster's etcds, so it definitely should *not* offer this as an option.